code/modules/atmospherics/machinery/components/binary_devices/circulator.dm

```diff
@@ -1,24 +1,32 @@
 //node2, air2, network2 correspond to input
 //node1, air1, network1 correspond to output
 
-
 /obj/machinery/atmospherics/components/binary/circulator
 	name = "circulator/heat exchanger"
 	desc = "A gas circulator pump and heat exchanger."
-	icon_state = "circ1-off"
+	icon_state = "circ-off"
 
-	var/side = CIRC_LEFT
-	var/status = 0
+	var/active = FALSE
 
 	var/last_pressure_delta = 0
 	pipe_flags = PIPING_ONE_PER_TURF
 
 	anchored = TRUE
 	density = TRUE
+	var/const/hot = 0
+	var/const/cold = 1
+
+	var/mode = hot
+	var/obj/machinery/power/generator/generator
 
-	var/global/const/CIRC_LEFT = 1
-	var/global/const/CIRC_RIGHT = 2
+//default cold circ for mappers
+/obj/machinery/atmospherics/components/binary/circulator/cold
+	mode = cold
 
+/obj/machinery/atmospherics/components/binary/circulator/Initialize(mapload)
+	.=..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/machine/circulator(null)

```

Comment:
not sure how safe this is, but it should be `component_parts = list(new /obj/item/circuitboard/machine/circulator)`

Reply:
🤷‍♂️ I just copied what the deep fryer does
https://github.com/yogstation13/Yogstation-TG/blob/6ea0e3e5edaaaf1dd8db51437ae72f77b197087f/code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm#L57